components/sup/src/event/nitox.rs

```diff
@@ -0,0 +1,90 @@
+            streaming::client::NatsStreamingClient,
+            NatsClient,
+            NatsClientOptions};
+use std::{sync::mpsc as std_mpsc,
+          thread};
+use tokio::{executor,
+            runtime::current_thread::Runtime as ThreadRuntime};
+/// All messages are published under this subject.
+const HABITAT_SUBJECT: &str = "habitat";
+
+pub(super) fn init_stream(conn_info: EventConnectionInfo) -> Result<EventStream> {
+    // TODO (CM): Investigate back-pressure scenarios
+    let (event_tx, event_rx) = futures_mpsc::unbounded();
+    let (sync_tx, sync_rx) = std_mpsc::sync_channel(0); // rendezvous channel
+
+    // TODO (CM): We could theoretically create this future and spawn
+    // it in the Supervisor's Tokio runtime, but there's currently a
+    // bug: https://github.com/YellowInnovation/nitox/issues/24
+
+    thread::Builder::new().name("events".to_string())
+                          .spawn(move || {
+                              let EventConnectionInfo { name,
+                                                        verbose,
+                                                        cluster_uri,
+                                                        cluster_id,
+                                                        auth_token, } = conn_info;
+                              let cc =
+                                  ConnectCommand::builder().name(Some(name))
+                                                           .verbose(verbose)
+                                                           .auth_token(Some(auth_token.to_string()))
+                                                           .tls_required(false)
+                                                           .build()
+                                                           .expect("Could not create NATS \
+                                                                    ConnectCommand");
+                              let opts =
+                                  NatsClientOptions::builder().connect_command(cc)
+                                                              .cluster_uri(cluster_uri.as_str())
+                                                              .build()
+                                                              .expect("Could not create \
+                                                                       NatsClientOptions");
+
+                              let publisher = NatsClient::from_options(opts).map_err(|e| {
+                                                  error!("Error connecting to NATS: {}", e);
+                                                  e.into()
+                                              })
+                                              .and_then(|client| {
+                                                  NatsStreamingClient::from(client)
+                        .cluster_id(cluster_id)
+                        .connect()
+                                              })
```

Comment:
Not quite sure what's up with the formatting here, but this is all really:

```rust
...
.and_then(|client| {
    NatsStreamingClient::from(client)
        .cluster_id(cluster_id)
        .connect()
})
...
```
The `and_then` is a future combinator that takes a non-streaming client connection (created asynchronously) and then effectively upgrades it to a streaming client connection. The `NatsStreamingClient::from(...).cluster_id(...).connect()` call is a builder pattern which generates the _next_ future in the chain with the final call to `connect()`.

Reply:
So the above suggestion is a semantic (and undesirable) change?

It might be worth disabling rustfmt on this section (and possibly filing an issue with rustfmt) so that we can fix up the formatting by hand.

Reply:
Yes, your suggestion is a semantic (and in fact, non-compilable) change.

I'll try out `#[rustfmt::skip]` and see if that does what we want.

Reply:
First time in awhile I've gotten something to *build* that was wrong 😆

Reply:
Also, let's reference https://github.com/rust-lang/rustfmt/issues/1762 in the code there. I'm subscribed to the issue, so we can remove our `#[rustfmt::skip]` if it gets fixed later.

Reply:
I've got a link to that issue in the code comments, and I've subscribed as well.

I suspect the reason you got it to build is because this is in the Nitox code, and that's not compiled by default... you'd have to use `cargo build --feature nitox_stream` to run into the error.

At least, that's what I didn't do initially when I _also_ got it to build 😉🤣

Reply:
Ah, good point. Do we have CI which will build with all the different features?

Reply:
I will add one now 😄